service/service.py

```diff
@@ -93,55 +93,71 @@ def index():
+    'price': fields.Float(required=True,
+                          description='Price'),
     'text': fields.String(required=True,
-                              description='Name of the product'),
-    'state': fields.String(required=True,
-                              description='State of the product in shopcart.(ADDED:0 (Default), REMOVED:1, DONE:2)')
+                          description='Name of the product'),
+    'state': fields.Integer(required=True,
+                            description='State of the product in shopcart.(ADDED:0 (Default), REMOVED:1, DONE:2)')
 })
 
 create_model = api.model('Shopcart', {
-    'product_id': fields.String(required=True,
-                          description='Product Identifier'),
-    'customer_id': fields.String(required=True,
-                              description='Customer Identifier'),
-    'quantity': fields.Boolean(required=True,
+    'product_id': fields.Integer(required=True,
+                                 description='Product Identifier'),
+    'customer_id': fields.Integer(required=True,
+                                  description='Customer Identifier'),
+    'quantity': fields.Integer(required=True,
                                 description='Quantity of the product'),
-    'price': fields.String(required=True,
-                              description='Price'),
+    'price': fields.Float(required=True,
+                           description='Price'),
     'text': fields.String(required=True,
                               description='Name of the product')
 })
 
+shopcart_args = reqparse.RequestParser()
+shopcart_args.add_argument('price', type=float, required=False, help='List shocpart items (cheapter than target_price if there exists one)')
 ######################################################################
 # LIST ALL ITEMS IN ONE SHOP CART ---
 ######################################################################
-@app.route('/shopcarts/<int:customer_id>', methods=['GET'])
-def list_cart_iterms(customer_id):
-    """ Returns list of all of the shop cart items"""
-    if request.args.get('price') == None:
-        app.logger.info('Request to list all items in shopcart with customer_id: %s', customer_id)
-        items = []
-        items = Shopcart.find_by_customer_id(customer_id)
-        results = [item.serialize() for item in items]
-        return make_response(jsonify(results), status.HTTP_200_OK)
-
-    else:    
-        target_price = request.args.get('price')
-        app.logger.info('Request to query all items in shopcart with customer_id: %s', customer_id)
-        items = []
-        items = Shopcart.query_by_target_price(customer_id, target_price)
-        results = [item.serialize() for item in items]
-        return make_response(jsonify(results), status.HTTP_200_OK)
+@api.route('/shopcarts/<int:customer_id>', strict_slashes=False)
+@api.param('customer_id', 'Customer Identifier')
+class shopcartsCollection(Resource):
```

Comment:
I think we have no concept of collection so can we name it ShopcartItem? And the naming convention is to start with upper case